### PR TITLE
Update wait_for_unit to accept Juju 2.0 output

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/50/
Test PASSed.

**Comment by bac** [comment]
This approach is pretty brute force.

I looked at adding some more generic helper functions for handling `juju version` but discovered there is support in the latest version of charmhelpers.

We are on an old, unmaintained copy of charmhelpers.  I plan to change this charm to use the newest version and fix whatever fallout there is.

To QA this branch:
```
juju bootstrap
make deploy
```
with Juju 1 and 2.  The GUI should launch, `juju status` should show it up and running.  For juju 1 you should be able to login.  For juju 2 you'll get a login screen but cannot proceed past it. This is due to other issues with Juju 2 we are working on.

File: tests/helpers.py [hunk 1]
```diff
@@ -200,19 +200,24 @@ def wait_for_unit(sevice):
...
         if service is None or not service.get('exposed'):
             continue
         units = service.get('units', {})
         if not len(units):
             continue
         unit = units.values()[0]
-        state = unit['agent-state']
+        state = unit.get('agent-state')
+        if state is None:
+            state = unit['agent-status']['current']
```
Comment (frankban) [inline tests/helpers.py]: Add a comment that this code path is for Juju 2?

**Comment by frankban** [comment]
Code looks good, QAing now.

File: tests/helpers.py (hunk 1 above)
Comment (bac) [inline tests/helpers.py]: Good idea

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/51/
Test PASSed.

File: tests/helpers.py (hunk 1 above)
Comment (bac) [inline tests/helpers.py]: Done.

**Comment by frankban** [comment]
:+1: QA ok thank you!

**Comment by jaycee** [comment]
:+1: thanks @bac!

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**